ui/app/actions.js

```diff
@@ -669,12 +671,36 @@ function markAccountsFound() {
 // config
 //
 
+// default rpc target refers to localhost:8545 in this instance.
+function setDefaultRpcTarget (rpcList) {
+  log.debug(`background.setDefaultRpcTarget`)
+  background.setRpcTarget('http://localhost:8545')
+  return (dispatch) => {
+    dispatch({
+      type: actions.SET_RPC_TARGET,
+      value: 'http://localhost:8545',
+    })
+    dispatch({
+      type: actions.SET_RPC_LIST,
+      value: rpcList,
+    })
+  }
+}
+
 function setRpcTarget (newRpc) {
-  log.debug(`background.setRpcTarget`)
-  background.setRpcTarget(newRpc)
-  return {
-    type: actions.SET_RPC_TARGET,
-    value: newRpc,
+  return (dispatch) => {
+    log.debug(`background.setRpcTarget`)
+    background.setRpcTarget(newRpc)
+    background.updateFrequentRpcList(newRpc, (rpcList) => {
```

Comment:
The first argument in a callback is an `error` object.  The *second* argument should be your `rpcList`.  Did you test this? I'd be surprised if this worked, since you're just calling `nodeify` on the background promise-returning function.